src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -799,4 +799,129 @@
+    end
+    let(:create_distributions_other_reponame) { create_list(:distribution, 2, reponame: 'another_repon_name') }
+
+    before do
+      login user
+    end
+
+    context "without any distributions passed" do
+      before do
+        post :save_distributions, project: user.home_project
+      end
+
+      it { expect(flash[:success]).to eq("Successfully added repositories") }
+      it { is_expected.to redirect_to(action: :repositories, project: user.home_project) }
+      it { expect(assigns(:project).repositories.count).to eq(0) }
+    end
+
+    context "with a distribution but without target repository" do
+      it "Raises an ActiveRecord::RecordNotFound exception" do
+        expect do
+          post :save_distributions, project: user.home_project,
+                                    distributions: [distribution_without_target_repo.reponame]
+        end.to raise_error ActiveRecord::RecordNotFound
+      end
+    end
+
+    context "with two distributions using same repository" do
+      before do
+        request.env["HTTP_REFERER"] = root_url # Needed for the redirect_to :back
+        create_distributions_same_repo
+        post :save_distributions, project: user.home_project, distributions: [another_project_repository.name]
+      end
+
+      it do
+        expect(flash[:error]).to eq("Can't add repositories: Validation failed: Name #{another_project_repository.name} is already used by a " +
+                                       "repository of this project.")
+      end
+      it { is_expected.to redirect_to(:back) }
+    end
+
+    context "with a distribution properly set" do
+      before do
+        create_distributions_other_reponame
+        post :save_distributions, project: user.home_project, distributions: [distribution.reponame]
+      end
+
+      it { expect(flash[:success]).to eq("Successfully added repositories") }
+      it { is_expected.to redirect_to(action: :repositories, project: user.home_project) }
+      it { expect(assigns(:project).repositories.count).to eq(1) }
+      it { expect(assigns(:project).repositories.first.name).to eq(distribution.reponame) }
+    end
+
+    context "with a distribution called images" do
+      before do
+        Project.any_instance.stubs(:prepend_kiwi_config).returns(true)
+        post :save_distributions, project: user.home_project, images: true
+      end
+
+      it { expect(flash[:success]).to eq("Successfully added repositories") }
+      it { is_expected.to redirect_to(action: :repositories, project: user.home_project) }
+      it { expect(assigns(:project).repositories.count).to eq(1) }
+      it { expect(assigns(:project).repositories.first.name).to eq('images') }
+      it { expect(assigns(:project).repositories.first.repository_architectures.count).to eq(Architecture.available.count) }
+    end
+  end
+
+  describe 'GET #remove_target_request_dialog' do
```

Comment:
Just covering the method... should I remove this? Is a trivial thing I know, but at least this is covered by some test :disappointed: